server/bin/test_data_importer.py

```diff
@@ -0,0 +1,706 @@
+        return self.request('DELETE', endpoint, data, params, headers, content_type, accept_type)
+
+    def request(self, req_type, endpoint, data=None, params=None, headers=None, content_type='application/json', accept_type='application/json'):
+        if params is None: params = {}
+        if headers is None: headers = {}
+
+        # Add our content-type and accept type if they weren't explicitly provided in the headers already
+        if not ci_in(headers, 'Content-Type'): headers['Content-Type'] = content_type
+        if not ci_in(headers, 'Accept'): headers['Accept'] = accept_type
+
+        log.log(LOGLVL_TRACE, "Sending request: %s %s (params: %s, headers: %s, data: %s)" % (req_type, endpoint, params, headers, data))
+
+        response = requests.request(req_type, self.build_url(endpoint),
+            json=data,
+            auth=(self.username, self.password),
+            params=params,
+            headers=headers,
+            verify=False)
+
+        response.raise_for_status()
+        return response
+
+    def status(self):
+        if self.status_cache is None:
+            self.status_cache = self.get('status').json()
+
+        return self.status_cache
+
+    def determine_mode(self):
+        if self.hosted_test is None:
+            self.hosted_test = not self.status()['standalone']
+
+            if self.hosted_test:
+                try:
+                    self.get('hostedtest/alive', accept_type='text/plain')
+                except requests.exceptions.HTTPError:
+                    self.hosted_test = False
+
+        return self.hosted_test
+
+    # Candlepin specific functions
+    def create_owner(self, owner_data):
+        owner = {
+            'key': owner_data['name'],
+            'displayName': owner_data['displayName']
+        }
+
+        if 'contentAccessModeList' in owner_data:
+            owner['contentAccessModeList'] = owner_data['contentAccessModeList']
+
+        if 'contentAccessMode' in owner_data:
+            owner['contentAccessMode'] = owner_data['contentAccessMode']
+
+        owner = self.post('owners', owner).json()
+
+        # If we're operating in hosted-test mode, also create the org upstream for consistency
+        if self.hosted_test:
+            self.post('hostedtest/owners', { 'key': owner['key'] })
+
+        return owner
+
+    def create_user(self, user_data):
+        user = safe_merge(user_data, {}, include=['username', 'password'])
+        user['superAdmin'] = user_data['superadmin'] if 'superadmin' in user_data else False
+
+        return self.post('users', user).json()
+
+    def create_role(self, role_name, permissions):
+        role = {
+            'name': role_name,
+            'permissions': permissions
+        }
+
+        return self.post('roles', role).json()
+
+    def add_role_to_user(self, username, role_name):
+        return self.post("roles/%s/users/%s" % (role_name, username)).json()
```

Comment:
Using `"%s" % role_name` is considered as obsoleted. Please use "{var}".format(var=variable).

Reply:
Done.

Reply:
Thanks. It would be better to use f-strings, but I think that we still want to test it on RHEL/CentOS 7, right? BTW: PyCharm can do this conversation for you (shortcut Alt-Enter).